libensemble/worker.py

```diff
@@ -253,22 +253,27 @@ def _handle_calc(self, Work: dict, calc_in: npt.NDArray) -> (npt.NDArray, dict,
 
                 logger.debug(f"Returned from user function for {enum_desc} {calc_id}")
 
-            assert isinstance(out, tuple), "Calculation output must be a tuple."
-            assert len(out) >= 2, "Calculation output must be at least two elements."
-
-            if len(out) >= 3:
-                calc_status = out[2]
-            else:
-                calc_status = UNSET_TAG
-
+            calc_status = UNSET_TAG
             # Check for buffered receive
             if self.comm.recv_buffer:
                 tag, message = self.comm.recv()
-                if tag in [STOP_TAG, PERSIS_STOP]:
-                    if message is MAN_SIGNAL_FINISH:
-                        calc_status = MAN_SIGNAL_FINISH
+                if tag in [STOP_TAG, PERSIS_STOP] and message is MAN_SIGNAL_FINISH:
+                    calc_status = MAN_SIGNAL_FINISH
+
+            if out:  # better way of doing this logic?
+                if len(out) >= 3:  # Out, persis_info, calc_status
+                    calc_status = out[2]
+                    return out
+                elif len(out) == 2:  # Out, persis_info OR Out, calc_status
+                    if isinstance(out[1], int) or isinstance(out[1], str):  # got Out, calc_status
+                        calc_status = out[1]
+                        return out[0], Work["persis_info"], calc_status
+                    return *out, calc_status  # got Out, persis_info
+                else:
+                    return out, Work["persis_info"], calc_status
+            else:
+                return np.zeros(1), Work["persis_info"], calc_status
```

Comment:
What happens with this zero? Should it be None?

Reply:
Based on discussion, it probably _could_ be None, then the manager is responsible for slotting in the value